### DISCUSSION TITLE
Link prefetching only for SSG sites

### DISCUSSION THREAD

**Discussion by trm217** [vercel/next.js#41463] [Ideas]
According to the docs the link prefetch-prop works as follow:

> prefetch - Prefetch the page in the background. Defaults to true. Any <Link /> that is in the viewport (initially or through scroll) will be preloaded. Prefetch can be disabled by passing prefetch={false}. When prefetch is set to false, prefetching will still occur on hover. Pages using [Static Generation](https://nextjs.org/docs/basic-features/data-fetching/get-static-props) will preload JSON files with the data for faster page transitions. Prefetching is only enabled in production.

In our production site we noticed however that every page is prefetched no matter if it's SSG or SSR (or legacy getInitialProps). This in our case has caused quite a lot of issues with the prefetch behaviour changing over time.

I'd like it, if I was able to only enable prefetch for links that point to an SSG site and to defer the loading on SSR until the link is actually clicked (or hovered on, since that can't be disabled anyways).

**Comment by icyJoseph** [comment]
Hi,

Are you perhaps experiencing a bug where pages with `getServerSideProps` are being prefetched? This was fixed in `12.3.x`, but was present on the version right before it.

**Reply by trm217** [reply]
That's possible, I'll double check and get back to you. Thanks for the hint.

**Reply by trm217** [reply]
@icyJoseph Any chance you have a link to the bug description?
Just checked the change-logs and couldn't notice the mentioned issue.

**Reply by icyJoseph** [reply]
I mean, the JS will always be prefetched but the JSON data for the GSSP pages, should not, and that was happening in the `12.2.5` version, which caused some of our data to diverge between certain user states... 😱

**Reply by icyJoseph** [reply]
https://github.com/vercel/next.js/issues/38918 -> I derived from here that we had to upgrade to `12.3.x`

**Reply by trm217** [reply]
Our production site that crashed twice (due to high load) is using 12.2.4

**Reply by icyJoseph** [reply]
https://github.com/vercel/next.js/pull/39920 and I guess the MR for it

**Reply by icyJoseph** [reply]
Was it enough to upgrade to 12.3.x? Did it fix your issue?

**Reply by trm217** [reply]
We didn't upgrade it, and I won't for quite some time. Almost all Next.js updates, no matter if it was a patch or minor release, have introduced bugs, performance issues or "breaking "changes the last few times.
There for I am now extremely hesitant to update everything Next.js related. :-/

**Reply by icyJoseph** [reply]
Understandable. However 12.3.x has been super stable for us, and every new version more tests are added to Next.js which means more resilient performance. 

Truth be told, we are a bit bolder with keeping Next.js up-to-date because on our current project we have two pre-prod environments, so buggy versions have a really hard time reaching production.

**Reply by trm217** [reply]
We also have test systems, but I lost a lot of trust in Next.js test-suite. Many bugs went through that we can't catch without a large additional hassle, but I would've expected Next.js to ensure things work as intended.
For example the pre-compilation of IE-polyfills was broken for quite a while without it being fixed. 
Additionally many issues are only really visible once a production site is under heavy load. And I just can't keep up with checking every Next.js feature on every single update I make. Therefor I'd rather only update if it's really necessary.
